### PR TITLE
Add source and debuginfo to main repo files

### PR DESCRIPTION
**Pull request by tofay**
Signed-off-by: Tom Fay <tomfay@microsoft.com>

###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

Add source/debuginfo repo configuration, disabled, to the official and extended repository definitions.
This enables integration with `dnf download` for managing source and debuginfo packages, and is the approach taken by other RPM based distros.

###### Does this affect the toolchain?  

YES. I've not run the the whole rebuild as I understand that takes several hours.

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

Tested locally with `dnf download`.
There's a quirk with `dnf download --debuginfo` where if you have installed `mariner-repos-debug` already and you now run `dnf download --debuginfo` then you get duplicates. That seems sufficiently niche and easily resolvable when encountered (by uninstalling mariner-repos-debug).
`mariner-repos-*debug` arguably needn't exist now.

### CODE REVIEW

File: SPECS/mariner-repos/mariner-official-base.repo [hunk 1]
```diff
@@ -7,3 +7,23 @@ repo_gpgcheck=1
...
+name=CBL-Mariner Official Base Sources
+baseurl=https://packages.microsoft.com/cbl-mariner/$releasever/prod/base/srpms/
+gpgkey=file:///etc/pki/rpm-gpg/MICROSOFT-RPM-GPG-KEY file:///etc/pki/rpm-gpg/MICROSOFT-METADATA-GPG-KEY
+gpgcheck=1
+repo_gpgcheck=1
+enabled=0
+skip_if_unavailable=True
+sslverify=1
+
+[mariner-official-base-debuginfo]
```
Comment (anphel31) [inline SPECS/mariner-repos/mariner-official-base.repo:21]: Is this part needed, since it's already available in mariner-debuginfo.repo?

Comment (tofay) [reply SPECS/mariner-repos/mariner-official-base.repo:21]: It matches what other RPM based distributions do, which means:
- `dnf debuginfo-install` and `dnf download --debuginfo` work like other RPM based distributions (they enable the debuginfo repos)
- articles like https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/6/html/developer_guide/intro.debuginfo work for mariner too

File: SPECS/mariner-repos/mariner-repos.spec [hunk 2]
```diff
@@ -214,6 +214,10 @@ gpg --batch --yes --delete-keys 2BC94FFF7015A5F28F1537AD0CD9FED33135CE90
 %{_sysconfdir}/pki/rpm-gpg/MICROSOFT-METADATA-GPG-KEY
 
 %changelog
+* Fri Jul 15 2022 Tom Fay <tomfay@microsoft.com> - 2.0-8
+- Add disabled and source repo configuration to official and extended repo definition.
```
Comment (christopherco) [inline SPECS/mariner-repos/mariner-repos.spec:218]: nit
```suggestion
- Add debuginfo and source repo configuration to official and extended repo definition.
```

**Review by christopherco — CHANGES_REQUESTED** [review]
Thank you for contributing this change! Please update the target branch to `main` so it hits our main development branch first. Then it will get merged into the 2.0 release branch when the next release happens.

**Review by christopherco — APPROVED** [review]

**Comment by tofay** [comment]
Superseded by https://github.com/microsoft/CBL-Mariner/pull/3378

**Pull request closed without merging**